tests/bootstrap_project/test/project_test.dart

```diff
@@ -0,0 +1,432 @@
+                serverDir,
+                'generated',
+                'migration',
+                'migrations',
+                projectName,
+              ),
+            ).existsSync(),
+            isTrue,
+            reason: 'Server migrations folder does not exist.',
+          );
+        });
+
+        test('has project migration registry', () {
+          expect(
+            File(path.join(
+              serverDir,
+              'generated',
+              'migration',
+              'migrations',
+              projectName,
+              'migration_registry.json',
+            )).existsSync(),
+            isTrue,
+            reason: 'Server migration registry does not exist.',
+          );
+        });
+      });
+
+      group('then the flutter project', () {
+        test('folder is created', () {
+          expect(
+            Directory(flutterDir).existsSync(),
+            isTrue,
+            reason: 'Flutter folder does not exist.',
+          );
+        });
+
+        test('has a pubspec file', () {
+          expect(
+            File(path.join(flutterDir, 'pubspec.yaml')).existsSync(),
+            isTrue,
+            reason: 'Flutter pubspec file does not exist.',
+          );
+        });
+
+        test('has a main file', () {
+          expect(
+            File(path.join(flutterDir, 'lib', 'main.dart')).existsSync(),
+            isTrue,
+            reason: 'Flutter main file does not exist.',
+          );
+        });
+      });
+
+      group('then the client project', () {
+        test('folder is created', () {
+          expect(
+            Directory(clientDir).existsSync(),
+            isTrue,
+            reason: 'Client folder does not exist.',
+          );
+        });
+
+        test('has a pubspec file', () {
+          expect(
+            File(path.join(clientDir, 'pubspec.yaml')).existsSync(),
+            isTrue,
+            reason: 'Client pubspec file does not exist.',
+          );
+        });
+
+        test('has a project_client file', () {
+          expect(
+            File(
+              path.join(clientDir, 'lib', '${projectName}_client.dart'),
+            ).existsSync(),
+            isTrue,
+            reason: 'Client project_client file does not exist.',
+          );
+        });
+
+        test('has a protocol client file', () {
+          expect(
+            File(
+              path.join(clientDir, 'lib', 'src', 'protocol', 'client.dart'),
+            ).existsSync(),
+            isTrue,
+            reason: 'Client protocol client file does not exist.',
+          );
+        });
+      });
+    });
+  }, timeout: timeout);
+
+  group('Given a clean state', () {
+    final (projectName, commandRoot) = createRandomProjectName(tempPath);
+    final (serverDir, _, clientDir) = createProjectFolderPaths(projectName);
+
+    late Process createProcess;
+
+    tearDown(() async {
+      print('----------------------------------------------------------');
```

Comment:
Question - Should this print be removed?

Reply:
yes